code/processes/tickerlogreplay.q

```diff
@@ -337,31 +352,35 @@ getlogfile:{
   enlist hsym f
  };
 
-// Return contents of log directory if it exists. If segmented use meta table, if not, grab directory contents
+// Return contents of log directory if it exists
 getlogdir:{
   if[()~key hsym d:.replay.tplogdir;.lg.e[`getlogdir;m:"Specified log directory ",string[d]," does not exist"];'m];
+  if[d like "*.gz";.lg.e[`getlogdir;m:"Zipped log directories not supported."];'m];
   $[.replay.segmentedmode;.replay.getstplogs[d];.Q.dd[logdir;] each key logdir:hsym d]
  };
 
 // Use STP meta table and tplogdir to build log names
 getstplogs:{[logdir]
-  metatable:@[get;.Q.dd[hsym logdir;`stpmeta];{.lg.e[`getstpmeta;m:"Log directory must contain valid STP meta table"];'m}];
+  // If meta table is zipped, assume all other logs are zipped as well and build log names accordingly
+  if[z:`stpmeta.gz in key d:hsym logdir;system "gunzip ",1_string .Q.dd[d;`stpmeta.gz]];
```

Comment:
```suggestion
  if[z:`stpmeta.gz in key d:hsym logdir;system"gunzip ",1_string .Q.dd[d;`stpmeta.gz]];
```

the `system"gunzip ...` probably means this bit won't work on Windows either I think? might need a check for that and/or error trapping here incase the unzip fails etc.

also, it might be possible to avoid unzipping & later rezipping if you can use named pipes & `zcat` to read the file similar to https://github.com/AquaQAnalytics/TorQ-CME/blob/master/code/processes/cmedecoder.q#L35-L36. Not 100% sure if `get` works with named pipes though

Reply:
Yeah I tried this and you can get the binary contents out via a named pipe but `get` seems to add some kind of header to the binary, so if you `-9!` the pipe output it doesn't recognise it as a q object. Any idea if this can be gotten round?

Reply:
If it doesn't work with a simple `get`, probably not worth it tbh